client/client-lib/src/lib.rs

```diff
@@ -619,14 +685,36 @@ impl Client<UserClientConfig> {
             .submit_tx_with_change(tx, DbBatch::new(), &mut rng)
             .await?;
 
+        if let Some(mut saved_payment) = self.ln_client().fetch_payment(ci.invoice.payment_hash()) {
+            saved_payment.timestamp = duration_since_epoch().as_millis();
+            match saved_payment.detail {
+                TypedPaymentDetail::InboundLightning {
+                    invoice,
+                    used_gateway,
+                    ..
+                } => {
+                    saved_payment.detail = TypedPaymentDetail::InboundLightning {
+                        invoice: (invoice),
+                        paid: true,
+                        used_gateway: (used_gateway),
+                    }
+                }
+                _ => panic!("retrieved payment history is of wrong type! (LN)"),
+            }
+            self.ln_client().save_payment(&saved_payment);
+        }
```

Comment:
Updating of redundant data seems risky. We might forget to update payments when e.g. introducing a new failure case. I'd much rather only save the payment details once and then use other DB entries to determine payment state (even though that's a bit less efficient, but we'd paginate anyway I assume).